#163388137 Enable a user to create a profile

dev-jey opened the pull request:
#### What does this PR do?
It enables a user to create, view and update their profile.
#### Description of Task to be completed?
- write tests for the create and get profile functionality
- create functionality for a user to create and only view their own profile
- create functionality for a user to view all other user's profiles
- Ensure that a user can update their profile
- Upload profile picture to cloudinary
#### How should this be manually tested?
1. Clone the repo
```
git clone https://github.com/andela/ah-the-phoenix.git
```
2. Change it into the current directory
```
cd ah-the-phoenix
```
3. Checkout to the Feature branch
```    
git checkout ft-create-user-profile-163388137
```
4. Run the server
`python manage.py runserver`

5.   1. Send a `get` request to the url below to view the created profile.
          Remember to append the registered username at the end of the url
         `http://127.0.0.1:8000/api/v1/profiles/<username>/`

       2. Send a `patch` request to the same url with the following keys and relevant values
         `Bio`
        `image`

6. Send a `get` request to the url below to get all profiles.
`http://127.0.0.1:8000/api/v1/profiles/`


#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[#163388137 ](https://www.pivotaltracker.com/story/show/163388137)

#### Screenshots
1. Signup
<img width="978" alt="screen shot 2019-02-13 at 11 42 07" src="https://user-images.githubusercontent.com/42576922/52698609-e1ae5d00-2f84-11e9-8343-22718a56d04f.png">
2. Login
<img width="980" alt="screen shot 2019-02-13 at 11 42 46" src="https://user-images.githubusercontent.com/42576922/52698629-ec68f200-2f84-11e9-8557-e2ac6f5b3d16.png">
3. Get profile
<img width="974" alt="screen shot 2019-02-13 at 11 44 16" src="https://user-images.githubusercontent.com/42576922/52698643-f5f25a00-2f84-11e9-871a-7e6952bae0b9.png">
4. Update profile
<img width="978" alt="screen shot 2019-02-13 at 11 44 50" src="https://user-images.githubusercontent.com/42576922/52698658-03a7df80-2f85-11e9-8f5e-020127d4a7f8.png">
5. Get all profiles
<img width="978" alt="screen shot 2019-02-13 at 11 48 05" src="https://user-images.githubusercontent.com/42576922/52698773-44075d80-2f85-11e9-878c-addd2a36cbeb.png">

dev-jey committed:
feature(profiles): enable a user to create a profile

- write tests for the create and get profile functionality
- create functionality for a user to create and only view their own profile
- create functionality for a user to view all other user's profiles
- Ensure that a user can update their profile
- Upload profile picture to cloudinary

[Finishes #163388137]

asirvex commented on authors/apps/profiles/urls.py:
```diff
@@ -0,0 +1,10 @@
+
+urlpatterns = [
+    path('profiles/', ProfileGetAPIView.as_view(), name='get_profiles'),
+    path('profiles/<str:username>/', ProfileUpdateAPIView.as_view(), name='user_profile')
```
This line is too long. Try shortening it

dev-jey replied:
I have not yet added flake8 standards because the `tox PR` isn't merged into `develop`. When it's merged, I will rebase and correct all the linting errors available.

dev-jey committed:
feature(profiles): enable a user to create a profile

- work on feedback to incorporate the profiles functionality into the authentication app

[Finishes #163388137]

sirghiny commented on authors/apps/authentication/views.py:
```diff
@@ -276,3 +278,48 @@ def create(self, request):
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = ProfilesSerializer
+
+    def get(self, request, username, *args, **kwargs):
```
To uniquely identify a user, use the `pk`, not the username.

dev-jey committed:
feature(profiles): Enable a user to create a profile

- work on feedback to change the put method to patch
- change url to refer to pk instead of username

[Finishes #163388137]
